### DISCUSSION TITLE
Deploy on railway

### DISCUSSION THREAD

**Discussion by wilsntn** [dani-garcia/vaultwarden#4373] [Q&A]
Hi guys, any of you had an experience starting a vaultwarden service on railway? is there an option to deploy docker images on there but im getting the persistent volume error, but i already created a volume and atached to the service..

![Screenshot from 2024-02-21 10-00-35](https://github.com/dani-garcia/vaultwarden/assets/91102332/1fe4a373-519a-4bd3-8861-10634f96ab66)

and the documentation says

![image](https://github.com/dani-garcia/vaultwarden/assets/91102332/6664aecb-45f3-4765-844f-ba4953cca912)

is there any way to solve this problem?

**Comment by BlackDex** [comment]
Yes, set the `I_REALLY_WANT_VOLATILE_STORAGE` variable to `true`.
If that service somehow does not name the volume, it's difficult for us to detect this.

Via Docker/Podman or K8s those attached volumes all have a specific name isntead of a generated `uuid` if you defined it your self.
Probably this Railway has some other way of generating this, but I'm not sure of course.
It could very well be that there really isn't a persistent volume, which would mean all data will be lost once the container is stopped/re-created.

**Reply by wilsntn** [reply]
I have the volume name and i can create one and give the path but only after the service starts. 
And i guess the image looks for a volume in the build time right?

**Reply by BlackDex** [reply]
No, it looks at the mount points in runtime.

**Comment by wiasliaw** [comment]
I had set the `DATABASE_URL` in env but got the same error.
It seems that it cannot or unable to detect the storage in runtime.
